### ISSUE TITLE
The function of disabling strict HTTP parsing in http request node does not work

### ISSUE THREAD

**Issue by thkfighter** [node-red/node-red#4217]
### Current Behavior

We have a software with a JSON-RPC server and maybe there is something not compliant with a strict HTTP header specification. When I use Node-RED HTTP request, the JSON-RPC server can response as normal, which can be seen through Wireshark, but Node-RED reject the response and give two errors, 
the first from http request node being "RequestError: Parse Error: Invalid header value char", 
the second from the debug node following the http request node, "RequestError: Parse Error: Invalid header value char : http://172.17.0.1:8080".
HPE_INVALID_HEADER_TOKEN is displayed below the http request node.

However, a python program using module requests works well with the JSON-RPC server.

### Expected Behavior

It would be great if the function of disabling strict HTTP parsing in http request node work.

### Steps To Reproduce

Sorry, we can not provide the relevant JSON-RPC server here.
Below is the response header from the JSON-RPC server, I cannot find anything suspicious.
```
HTTP/1.1 200 OK
Content-Type: application/json
Date: Sat, 10 Jun 2023 09:16:45 GMT
Connection: keep-alive
Cache-Control: no-cache, no-store, must-revalidate, private, max-age=0
Pragma: no-cache
Expires: 0
Strict-Transport-Security: max-age=15552000
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 1; mode=block
X-Content-Type-Options: nosniff
Referrer-Policy: same-origin
Content-Length: 1232
```

### Example flow

```
[
    {
        "id": "f391abdf87839c07",
        "type": "tab",
        "label": "Flow 3",
        "disabled": false,
        "info": "",
        "env": []
    },
    {
        "id": "49942c24f0fedf42",
        "type": "debug",
        "z": "f391abdf87839c07",
        "name": "debug 14",
        "active": true,
        "tosidebar": true,
        "console": true,
        "tostatus": false,
        "complete": "payload",
        "targetType": "msg",
        "statusVal": "",
        "statusType": "auto",
        "x": 870,
        "y": 160,
        "wires": []
    },
    {
        "id": "0eb9c5b1d2e3656c",
        "type": "http request",
        "z": "f391abdf87839c07",
        "name": "",
        "method": "POST",
        "ret": "obj",
        "paytoqs": "ignore",
        "url": "http://172.17.0.1:8080",
        "tls": "",
        "persist": false,
        "proxy": "",
        "insecureHTTPParser": true,
        "authType": "",
        "senderr": false,
        "headers": [
            {
                "keyType": "Content-Type",
                "keyValue": "",
                "valueType": "application/json",
                "valueValue": ""
            }
        ],
        "x": 630,
        "y": 160,
        "wires": [
            [
                "49942c24f0fedf42"
            ]
        ]
    },
    {
        "id": "0b9bc05d2fb6291d",
        "type": "inject",
        "z": "f391abdf87839c07",
        "name": "",
        "props": [
            {
                "p": "payload"
            },
            {
                "p": "topic",
                "vt": "str"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "",
        "payload": "{\"id\":403,\"jsonrpc\":\"2.0\",\"method\":\"aboutModulesList\",\"params\":{\"query\":{}}}",
        "payloadType": "json",
        "x": 430,
        "y": 160,
        "wires": [
            [
                "0eb9c5b1d2e3656c"
            ]
        ]
    }
]
```

### Environment

- Node-RED version: v3.0.2
- Node.js version: v18.16.0 / v19.9.0 / v20.2.0
- npm version: v9.6.6
- Platform/OS: Ubuntu 20.04, Linux 5.15.0-73-generic x64 LE
- Browser: Firefox 114.0 (64-bit)

**Comment by Steve-Mcl** [comment]
I believe this is fixed in 3.1.0 (currently in beta)

@hardillb can you confirm this? I recall a related thread on the forum: https://discourse.nodered.org/t/how-to-send-insecurehttpparser-request-with-the-http-request-node/77162/14

**Comment by Steve-Mcl** [comment]
@thkfighter if you are in a position to test beta-3 and let us know that would be great.

**Comment by thkfighter** [comment]
> @thkfighter if you are in a position to test beta-3 and let us know that would be great.

@Steve-Mcl Unfortunately, I am not. Can I get the beta version and have a try?

**Comment by knolleary** [comment]
@thkfighter details of the beta in the forum: https://discourse.nodered.org/t/node-red-3-1-0-beta-3-released/78716

**Comment by thkfighter** [comment]
@Steve-Mcl , @knolleary , Thanks. Yes, Node-RED v3.1.0-beta.3 has fixed this problem, the function of disabling strict HTTP parsing in http request node does work. 
Test environment:
    Node-RED version: v3.1.0-beta.3
    Node.js version: v18.16.0
    npm version: v9.7.1
    Platform/OS: Linux 5.19.0-43-generic x64 LE, Ubuntu 22.04
    Browser: Firefox 113.0.1 (64-bit)

Postscript: I tested on Ubuntu 22.04 rather than Ubuntu 20.04 because my Ubuntu 20.04 broke.

**Comment by Steve-Mcl** [comment]
Good news. Thanks for confirming. 

I think we can close this since the fox will be included in the imminent release of 3.1.0 final in the coming days.
